pkg/test/nodeclass.go

```diff
@@ -36,8 +36,8 @@ func EC2NodeClass(overrides ...v1beta1.EC2NodeClass) *v1beta1.EC2NodeClass {
 			panic(fmt.Sprintf("Failed to merge settings: %s", err))
 		}
 	}
-	if options.Spec.AMIFamily == nil {
-		options.Spec.AMIFamily = &v1beta1.AMIFamilyAL2
+	if len(options.Spec.AMISelectorTerms) == 0 {
+		options.Spec.AMISelectorTerms = []v1beta1.AMISelectorTerm{{Alias: "al2@latest"}}
```

Comment:
might as well upgrade?
```suggestion
		options.Spec.AMISelectorTerms = []v1beta1.AMISelectorTerm{{Alias: "al2023@latest"}}
```

Reply:
Fair, shouldn't make a difference. If it causes a bunch of other cascading issues based on assumptions tests have around al2 being the AMI family then I think we should decouple, but if it's straightforward might as well.

Reply:
Didn't make it into this PR, let's track this as a follow-up.